File: UI/rbike_mobile/lib/models/equipment.dart
import 'package:json_annotation/json_annotation.dart';

part 'equipment.g.dart';

@JsonSerializable()
class Equipment {
  int? equipmentId;
  String? name;
  String? description;
  double? price;
  String? image;
  String? status;
  int? stockQuantity;
  int? equipmentCategoryId;
  String? equipmentCategoryName;

  Equipment({
    this.equipmentId,
    this.name,
    this.description,
    this.price,
    this.image,
    this.status,
    this.stockQuantity,
    this.equipmentCategoryId,
    this.equipmentCategoryName,
  });

  factory Equipment.fromJson(Map<String, dynamic> json) => _$EquipmentFromJson(json);
  Map<String, dynamic> toJson() => _$EquipmentToJson(this);
} 

File: rBike.Model/Equipment.cs
using System;
using System.Collections.Generic;

namespace rBike.Model
{
    public class Equipment
    {
        public int EquipmentId { get; set; }

        public string Name { get; set; } = null!;

        public string? Description { get; set; }

        public decimal Price { get; set; }

        public byte[]? Image { get; set; }

        public string Status { get; set; } = null!;

        public int StockQuantity { get; set; }

        public int EquipmentCategoryId { get; set; }

        public string? EquipmentCategoryName { get; set; }

        public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
